asdf/tagged.py

```diff
@@ -98,6 +98,15 @@ def __eq__(self, other):
                 self._tag == other._tag)
 
 
+class TaggedTime(Tagged, time.Time):
+    """
+    An Astropy time object with a tag attached.
+    """
+    def __new__(cls, instance, tag):
+        self = time.Time.__new__(type(instance), instance)
+        self._tag = tag
+        return self
```

Comment:
I'm not quite sure what this is doing.  It will never return a `TaggedTime` object per se, because it just creates a plain `Time` object and adds a `._tag` attribute to it.  I admit the implementations of the other `Tagged` types is a bit odd too, but it seems the intent is to act as a wrapper around some type for internal use, rather than a fully functioning instance of that type.

Meanwhile if you wanted to make a `TaggedTime` that is the same as a `Time` object except for having the `._tag` attribute you would call `Time.__new__(cls, instance)`, and I think that should do it.